tools/vendor_hid_test.py

```diff
@@ -0,0 +1,259 @@
+"""These tests verify the functionality of the VendorHID interface."""
+import hid
+import time
+import unittest
+
+_OPENSK_VID = 0x1915
+_OPENSK_PID = 0x521F
+_FIDO_USAGE_PAGE = 0xF1D0
+_VENDOR_USAGE_PAGE = 0xFF00
+_PACKETS = 4
+_PACKET_SIZE = 64
+_SEND_DATA_SIZE = _PACKET_SIZE + 1
+_DEFAULT_CID = bytes([0xFF, 0xFF, 0xFF, 0xFF])
+
+
+def sleep():
+  #time.sleep(.01)
+  pass
+
+
+def ping_data_size(packets):
+  return 57 + 59 * (packets - 1)
+
+
+_BYTE_VAL = 0x10
+
+
+def get_byte():
+  global _BYTE_VAL
+  _BYTE_VAL = _BYTE_VAL + 1
+  return _BYTE_VAL
+
+
+class HidDevice(object):
+  """Class that helps interact with HID devices."""
+
+  def __init__(self, device):
+    self.device = device
+    self.dev = None
+    self.cid = None
+    self.rx_packets = []
+    self.create_and_init()
+
+  def __del__(self):
+    if self.dev:
+      self.dev.close()
+
+  def create_and_init(self) -> None:
+    self.dev = hid.Device(path=self.device['path'])
+    # Nonce is all zeros, because we don't care.
+    init_packet = [0] + list(_DEFAULT_CID) + [0x86, 0x00, 0x08] + [0x00] * 57
+    if len(init_packet) != _SEND_DATA_SIZE:
+      raise Exception(
+          f'Expected packet to be {_SEND_DATA_SIZE} but was {len(init_packet)}')
+    self.dev.write(bytes(init_packet))
+    self.cid = self.dev.read(_PACKET_SIZE, 2000)[15:19]
+    sleep()
+
+  def ping_init(self, packets=1, byte=0x88) -> int:
+    size = ping_data_size(packets)
+    ping_packet = [0] + list(self.cid) + [0x81, size // 256, size % 256
+                                         ] + [byte] * 57
+    if len(ping_packet) != _SEND_DATA_SIZE:
+      raise Exception(
+          f'Expected packet to be {_SEND_DATA_SIZE} but was {len(ping_packet)}')
+    r = self.dev.write(bytes(ping_packet))
+    sleep()
+    return r
+
+  def ping_continue(self, num, byte=0x88) -> int:
```

Comment:
Type hinting is not used consistently across the file?

Reply:
Fixed.